api/primary.go

```diff
@@ -122,16 +124,22 @@ func NewPrimary(cluster cluster.Cluster, tlsConfig *tls.Config, status StatusHan
 	// The new eventsHandler is initialized with no writers or channels.
 	// This is in api/events.go
 	eventsHandler := newEventsHandler()
+	// eventsQueue uses the watch package from SwarmKit, which is based on
+	// the go-events package. See https://github.com/docker/swarm/issues/2718
+	// for context
+	eventsQueue := watch.NewQueue()
```

Comment:
Now that https://github.com/docker/swarmkit/pull/2285 is merged, let's revendor swarmkit and change this constructor to:
```
NewQueue(WithTimeout(10 * time.Second), WithLimit(10000), WithCloseOutChan())
```

We probably want to define these values as config vars and plumb them through the standard config model used by classic swarm. 
Assuming 1KB per event (gross overestimation), the size limit of `10000` for the queue should occupy no more than 10MB per listener in memory. For the edge case of 100 simultaneous inefficient listeners, this would lead up to to an additional 1 GB of memory